src/ir/proto_to_ir.h

```diff
@@ -33,6 +34,20 @@ class ProtoToIR {
     std::unique_ptr<Module> module;
     std::unique_ptr<SsaNames> ssa_names;
   };
+
+  static Result Convert(IRContext& context,
+                        const std::string irTranslationUnitJsonProto) {
+      proto::IrTranslationUnit irTranslationUnit;
+      auto status = google::protobuf::util::JsonStringToMessage(irTranslationUnitJsonProto,
+                                                                &irTranslationUnit);
+      if (!status.ok()) {
+          LOG(ERROR) << "Json IR proto is corrupt " << status;
+          // TODO(cromwellian): handle this
```

Comment:
J, maybe I return optional<Result> or add a status field to Result

Reply:
Returning an `absl::StatusOr<Result>` is probably the best way forward.